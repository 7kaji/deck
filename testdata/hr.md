Page 1 - inside code block

```
---
code content
```

---

Page 2 - normal page

---

Page 3 - inside tilde code block

~~~~
---
code content
~~~~

---

Page 4 - inside 4 character fenced code block

````
---
````

~~~~
---
~~~~

---

Page 5 - not a horizontal rule but a list item

- item 1
- ---
- item 3

---

Page 6 - horizontal rule but not a page delmiter

-----

Page 7 - 5 dashes hr as a page delimiter

---

Page 8 - various horizontal rules which should not be page delimiters

- - - - -
*****
___

---

Page 9 - Setext heading

H2 heading
---

This should not be split because the above --- is a setext heading
