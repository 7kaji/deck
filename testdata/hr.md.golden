[
  {
    "layout": "",
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Page 1 - inside code block"
              }
            ]
          }
        ]
      }
    ],
    "code_blocks": [
      {
        "content": "---\ncode content\n"
      }
    ]
  },
  {
    "layout": "",
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Page 2 - normal page"
              }
            ]
          }
        ]
      }
    ]
  },
  {
    "layout": "",
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Page 3 - inside tilde code block"
              }
            ]
          }
        ]
      }
    ],
    "code_blocks": [
      {
        "content": "---\ncode content\n"
      }
    ]
  },
  {
    "layout": "",
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Page 4 - inside 4 character fenced code block"
              }
            ]
          }
        ]
      }
    ],
    "code_blocks": [
      {
        "content": "---\n"
      },
      {
        "content": "---\n"
      }
    ]
  },
  {
    "layout": "",
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Page 5 - not a horizontal rule but a list item"
              }
            ]
          },
          {
            "fragments": [
              {
                "value": "item 1"
              }
            ],
            "bullet": "-"
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "item 3"
              }
            ],
            "bullet": "-"
          }
        ]
      }
    ]
  },
  {
    "layout": "",
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Page 6 - horizontal rule but not a page delmiter"
              }
            ]
          }
        ]
      }
    ]
  },
  {
    "layout": "",
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Page 7 - 5 dashes hr as a page delimiter"
              }
            ]
          }
        ]
      }
    ]
  },
  {
    "layout": "",
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Page 8 - various horizontal rules which should not be page delimiters"
              }
            ]
          }
        ]
      },
      {}
    ]
  },
  {
    "layout": "",
    "titles": [
      "H2 heading"
    ],
    "bodies": [
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "Page 9 - Setext heading"
              }
            ]
          }
        ]
      },
      {
        "paragraphs": [
          {
            "fragments": [
              {
                "value": "This should not be split because the above --- is a setext heading"
              }
            ]
          }
        ]
      }
    ],
    "headings": {
      "2": [
        "H2 heading"
      ]
    }
  }
]